_posts/en/newsletters/2019-02-26-newsletter.md

```diff
@@ -0,0 +1,156 @@
+  spending or current balance via the change addresses.
+
+- [Taproot and scriptless scripts both use Schnorr, but how are they
+  different?]({{bse}}84086) In separate answers, Gregory Maxwell and
+  Andrew Chow each describe the differences between these two proposed
+  uses of Schnorr-based signatures.  Also includes a description of
+  adapter signatures, which can be used to enhance the efficiency and
+  privacy of trustless contract protocols.
+
+- [How much of block propagation time is used in verification?]({{bse}}84045)
+  Gregory Maxwell explains that it's probably closer to 0% than 1% for
+  the normal case, but that it can be much larger for a worst case block
+  that was specifically constructed to take a long time to verify.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].*
+
+- [Bitcoin Core #15348][] adds a [productivity hints][] document
+  describing tools and techniques developers have found to improve their
+  efficiency.  Although some are specific to Bitcoin Core and C++
+  development, others more generally apply to anyone who develops with
+  git or GitHub.
+
+- [LND #2419][] creates a time-based buffer for certain objects stored
+  in memory that allows them to be freed after a certain period of
+  non-use.  Previously, some unused objects were being kept in memory
+  indefinitely, leading to user-reported crashes when the program
+  exceeded the system's free memory.
+
+- [C-Lightning #2343][] provides the project's existing documentation in
+  a nicer format at [ReadTheDocs.io][cl rtd].
+
+- [C-Lightning #2353][], [#2360][C-Lightning #2360], and
+  [#2365][C-Lightning #2365] updates various RPCs to now accept values
+  suffixed with "btc", "sat", or "msat" to indicate the denomination of
+  the value.  The "btc" value allows up to 11 decimal places and the
+  "sat" value up to 3 decimal places but, in both cases, the last three
+  of these places must be zeroes for onchain operations where the extra
+  precision isn't supported by the Bitcoin protocol.  Other RPCs return
+  new fields ending in `_msat` that always contain the millisatoshi
+  value.  Several internal API changes are also made.
+
+- [C-Lightning #2380][] requires transactions have at least one
+  confirmation before the wallet will attempt to spend their bitcoins by
+  default.  This fixes a problem where the wallet would attempt to spend
+  its own unconfirmed change outputs but those payments would sometimes
+  get stuck because the earlier payments weren't confirming quickly.
+  Several RPCs related to payments receive a `minconf` parameter that
+  defaults to `1` but can be set to `0` to continue the old behavior or
+  set to a higher value if desired.
+
+- [Eclair #821][] adds more than a dozen heuristics used to help find
```

Comment:
I only count three heuristics: `cltvDeltaFactor`, `ageFactor` and `capacityFactor`.